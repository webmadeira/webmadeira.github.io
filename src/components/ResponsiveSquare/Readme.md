# Responsive Square

This component aims to be used to responsive containers that must resize according to the viewport size maintaining always the **same width and height**, forming a square.

## Usage 

~~~js
<ResponsiveSquare
  width="40%"
  minWidth="130px"
  style={{ ... }}
>
  <AnyContent />
</ResponsiveSquare>
~~~