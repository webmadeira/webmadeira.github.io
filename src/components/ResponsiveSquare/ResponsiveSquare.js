import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Square = styled.div`
  position: relative;
  width: ${props => props.width};
  min-width: ${props => props.minWidth};

  &:after {
    content: "";
    display: block;
    padding-bottom: 100%;
  }
`

const Content = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`

const ResponsiveSquare = ({
  children,
  width,
  minWidth,
  style,
}) => (
  <Square width={width} minWidth={minWidth} style={style}>
    <Content>
      {children}
    </Content>
  </Square>
)

ResponsiveSquare.propTypes = {
  children: PropTypes.node.isRequired,
  width: PropTypes.string.isRequired,
  minWidth: PropTypes.string.isRequired,
  style: PropTypes.shape({}),
}

ResponsiveSquare.defaultProps = {
  style: {},
}

export default ResponsiveSquare
