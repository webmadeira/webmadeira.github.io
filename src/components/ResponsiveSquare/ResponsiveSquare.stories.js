import React from 'react'
import { storiesOf } from '@storybook/react'
import withReadme from 'storybook-readme/with-readme'
import { withKnobs, text } from '@storybook/addon-knobs'
import Readme from './Readme.md'
import '../../App/global-styles'
import ResponsiveSquare from './ResponsiveSquare'


const style = {
  background: 'darkseagreen',
  boxShadow: '2px 2px 10px 2px',
  padding: '10px',
}

storiesOf('ResponsiveSquare', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(Readme))
  .add('default', () => (
    <ResponsiveSquare
      width={text('width', '40%')}
      minWidth={text('minWidth', '130px')}
      style={style}
    >
      <p>Resize me by resizing the window!</p>
    </ResponsiveSquare>
  ))
