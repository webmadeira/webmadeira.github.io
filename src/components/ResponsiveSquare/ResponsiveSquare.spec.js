import React from 'react'
import { shallow } from 'enzyme'
import ResponsiveSquare from './ResponsiveSquare'

describe('ResponsiveSquare component', () => {
  it('must render properly', () => {
    const component = shallow(
      (
        <ResponsiveSquare width="100%" minWidth="250px">
          <p>Some arbitrary content</p>
        </ResponsiveSquare>
      ),
    )

    expect(component).toMatchSnapshot()
  })
})
