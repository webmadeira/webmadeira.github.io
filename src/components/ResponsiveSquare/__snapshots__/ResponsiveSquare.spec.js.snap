// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveSquare component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResponsiveSquare
    minWidth="250px"
    style={Object {}}
    width="100%"
  >
    <p>
      Some arbitrary content
    </p>
  </ResponsiveSquare>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <p>
          Some arbitrary content
        </p>
      </styled.div>,
      "minWidth": "250px",
      "style": Object {},
      "width": "100%",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <p>
          Some arbitrary content
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Some arbitrary content",
        },
        "ref": null,
        "rendered": "Some arbitrary content",
        "type": "p",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <p>
            Some arbitrary content
          </p>
        </styled.div>,
        "minWidth": "250px",
        "style": Object {},
        "width": "100%",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            Some arbitrary content
          </p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some arbitrary content",
          },
          "ref": null,
          "rendered": "Some arbitrary content",
          "type": "p",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
